### ISSUE TITLE
Add  "No empty method" to  no-empty for ES6 methods

### ISSUE THREAD

**Issue by allenmanning** [eslint/eslint#4605]
It would check for methods like this

```
class Person

...

  getName () {
  }
```

**Comment by eslintbot** [comment]
@allenmanning Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by platinumazure** [comment]
What exactly defines "empty"? Should a comment make it non-empty, like in our empty block rule?

**Comment by mysticatea** [comment]
I think that to add an option which covers functions into [no-empty](http://eslint.org/docs/rules/no-empty) is fine.

**Comment by kaicataldo** [comment]
I'll work on this

**Comment by nzakas** [comment]
Agree, adding an option to `no-empty` that covers this case is fine. We probably want the option to be "methods" so that all functions aren't checked.

**Comment by mysticatea** [comment]
@nzakas To be clear, does that `"methods"` contain ES5 style methods and getter/setter?

```js
var obj = {
    foo: function() {},

    get foo2() {}
    set foo2(value) {}
};

obj.bar = function() {};

function Foo() {
    this.hello = function() {};
}

Foo.prototype.hello2 = function() {};
```

**Comment by mysticatea** [comment]
Ah, sorry, the title says "ES6 methods".

**Comment by nzakas** [comment]
We should keep "methods" to mean ES6 methods that are defined by shorthand notation in objects and classes.

**Comment by allenmanning** [comment]
Thank you.

One thing to consider as well is empty ES6 Constructors, or constructors that only call `super()`.
